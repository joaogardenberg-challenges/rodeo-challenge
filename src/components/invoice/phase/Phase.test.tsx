import { render, screen } from '@testing-library/react'
import { range } from 'lodash'

jest.mock('./CostItem')
jest.mock('./SubtotalPrice')

const Phase = require('./Phase').default

const costItemsCount = 5
const name = 'Phase name'

const props = {
  phase: {
    costItems: range(costItemsCount).map((index) => ({ id: index })),
    name
  }
}

describe('Phase', () => {
  it("Doesn't render when no phase", () => {
    render(<Phase />)
    const phaseContainer = screen.queryByTestId('phase-container')
    expect(phaseContainer).not.toBeInTheDocument()
  })

  it('Renders successfully', () => {
    const { container } = render(<Phase {...props} />)
    const phaseContainer = screen.getByTestId('phase-container')
    const phaseName = screen.getByTestId('phase-name')
    const costItems = screen.getAllByTestId('cost-item')

    expect(phaseContainer).toBeInTheDocument()
    expect(phaseName).toHaveTextContent(name)
    expect(costItems).toHaveLength(costItemsCount)
    expect(container).toMatchSnapshot()
  })
})
