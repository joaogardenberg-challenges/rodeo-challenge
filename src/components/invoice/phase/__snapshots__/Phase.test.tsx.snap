// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Phase Renders successfully 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
}

.c0 .phase-name {
  font-size: 1.25rem;
  text-align: center;
}

<div>
  <section
    class="c0 phase-section"
    data-testid="phase-container"
  >
    <h3
      class="phase-name"
      data-testid="phase-name"
    >
      Phase name
    </h3>
    <ol
      class="cost-items"
    >
      <li
        aria-hidden="true"
        class="cost-item cost-item--labels"
      >
        <span
          class="cost-item__name"
        >
          Name
        </span>
        <span
          class="cost-item__tax"
        >
          Tax
        </span>
        <span
          class="cost-item__description"
        >
          Description
        </span>
        <span
          class="cost-item__price"
        >
          Price
        </span>
      </li>
      <div
        data-testid="cost-item"
      />
      <div
        data-testid="cost-item"
      />
      <div
        data-testid="cost-item"
      />
      <div
        data-testid="cost-item"
      />
      <div
        data-testid="cost-item"
      />
      <div
        data-testid="subtotal-price"
      />
    </ol>
  </section>
</div>
`;
